nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/events/CheckAlreadyInColdStorageListener.java

```diff
@@ -84,17 +84,21 @@ public void handleEvent(Event event) {
         }
         Blob blob = (Blob) doc.getPropertyValue(FILE_CONTENT_PROPERTY);
         if (blob != null && blob instanceof ManagedBlob) {
+            ColdStorageService service = Framework.getService(ColdStorageService.class);
+            CoreSession session = docCtx.getCoreSession();
             if (ColdStorageHelper.isInColdStorage((ManagedBlob) blob)) {
+                // The blob already existed in ColdStorage state, let's update the document accordingly
                 log.debug("Main blob is already in cold storage, need to update document accordingly.");
-                ColdStorageService service = Framework.getService(ColdStorageService.class);
-                CoreSession session = docCtx.getCoreSession();
                 // We need to make sure the thumbnail is available to use it as placeholder of the main blob
                 thumbnailHelper.createThumbnailIfNeeded(session, doc);
                 doc = service.proceedMoveToColdStorage(session, doc.getRef());
                 if (doc.isVersion()) {
                     doc.putContextData(CoreSession.ALLOW_VERSION_WRITE, true);
                 }
                 session.saveDocument(doc);
+            } else {
+                // we may have force a restore by re-uploading a blob already in ColdStorage (see NXP-31830)
```

Comment:
maybe replace the comment with a log.debug

Reply:
> I understand that the blob is in s3 but with a wrong class and we want to fix invoking a restore, the comment should state that.

No, the creation of the doc may have overwritten the blob that had the storage class resulting in the blob having now the default storage class. We don't have a proper way to know and therefore we check that other documents referencing this blob needs to be updated accordingly (i.e. remove ColdStorage facet and set back main blob reference in `file:content` property)